src/node/desktop/src/native/desktop/desktop.cc

```diff
@@ -477,41 +477,37 @@ Napi::Value shortPathName(const Napi::CallbackInfo& info)
    // Get path as UTF-16 text.
    // Note that, on Windows, std::wstring and std::u16string are interchangable.
    // With the exception that char16_t is unsigned, and wchar_t is signed.
-   std::u16string u16Path = info[0].As<Napi::String>().Utf16Value();
-   std::wstring wPath(u16Path.begin(), u16Path.end());
+   std::u16string wPath = info[0].As<Napi::String>().Utf16Value();
 
    // Convert to a short path.
-   std::wstring wShortPath;
-   {
-      // https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getshortpathnamew
-      //
-      // If the lpszShortPath buffer is too small to contain the path, the
-      // return value is the size of the buffer, _in TCHARs_, that is required to
-      // hold the path _and the terminating null character_.
-      //
-      // Hence, the return value here does include a null terminator.
-      int requiredSizeInBytesIncludingTerminator = ::GetShortPathNameW(wPath.data(), nullptr, 0);
-      if (requiredSizeInBytesIncludingTerminator == 0) {
-         return path;
-      }
-
-      wchar_t* buffer = (wchar_t*) malloc(requiredSizeInBytesIncludingTerminator);
-      if (buffer == nullptr) {
-         return path;
-      }
+   //
+   // https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getshortpathnamew
+   //
+   // If the lpszShortPath buffer is too small to contain the path, the
+   // return value is the size of the buffer, _in TCHARs_, that is required to
+   // hold the path _and the terminating null character_.
+   int wShortPathSize = ::GetShortPathNameW((LPCWSTR) wPath.data(), nullptr, 0);
+   if (wShortPathSize == 0) {
+      return path;
+   }
 
-      int numBytesWritten = ::GetShortPathNameW(wPath.data(), buffer, requiredSizeInBytesIncludingTerminator);
-      if (numBytesWritten == 0) {
-         return path;
-      }
+   // Allocate space for the short path string.
+   LPWSTR wShortPath = (LPWSTR) malloc(wShortPathSize * sizeof(WCHAR));
+   if (wShortPath == nullptr) {
+      return path;
+   }
 
-      wShortPath.assign(buffer);
-      free(buffer);
+   // Copy into that buffer.
+   int numBytesWritten = ::GetShortPathNameW((LPCWSTR) wPath.data(), wShortPath, wShortPathSize);
+   if (numBytesWritten == 0) {
+      return path;
```

Comment:
Nit: memory leak (Real Code, none of those sissy smart pointers).

Reply:
Doh. This is why I prefer C++ -- the compiler is a lot better at remembering what memory I've allocated than I am!